### PR TITLE
Stop using API server's `--insecure-port`

### LINKED ISSUE #1283: Stop using API server's `--insecure-port`
From startup logs:
```
Flag --insecure-port has been deprecated, This flag will be removed in a future version.
```
@fabriziopandini @timothysc

### PR DESCRIPTION
**Pull request by RA489**
depreciate  API server's `--insecure-port`
fixes https://github.com/kubernetes/kubeadm/issues/1283

### CODE REVIEW

File: cluster/centos/master/scripts/apiserver.sh [hunk 1]
```diff
@@ -43,8 +43,8 @@ KUBE_ETCD_KEYFILE="--etcd-keyfile=/srv/kubernetes/etcd/client-key.pem"
 # --insecure-bind-address=127.0.0.1: The IP address on which to serve the --insecure-port.
 KUBE_API_ADDRESS="--insecure-bind-address=0.0.0.0"
 
-# --insecure-port=8080: The port on which to serve unsecured, unauthenticated access.
-KUBE_API_PORT="--insecure-port=8080"
+# --secure-port=8080: The port on which to serve unsecured, unauthenticated access.
```
Comment (MalloZup) [inline cluster/centos/master/scripts/apiserver.sh]: 2 nitpicks here:

1) `8080` port is used for http see: https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers
2)  The sentence `... to serve unsecured, unauth..`  need to be modified for `secure` mode

File: cluster/centos/master/scripts/apiserver.sh [hunk 2]
```diff
@@ -43,8 +43,8 @@ KUBE_ETCD_KEYFILE="--etcd-keyfile=/srv/kubernetes/etcd/client-key.pem"
...
+KUBE_API_PORT="--secure-port=8080"
```
Comment (MalloZup) [inline cluster/centos/master/scripts/apiserver.sh]: i think we need to think about the port numbers

File: staging/src/k8s.io/apiserver/pkg/server/options/deprecated_insecure_serving.go [hunk 3]
```diff
@@ -68,14 +68,14 @@ func (s *DeprecatedInsecureServingOptions) AddFlags(fs *pflag.FlagSet) {
...
+		"The IP address on which to serve the --secure-port (set to 0.0.0.0 for all IPv4 interfaces and :: for all IPv6 interfaces).")
 	fs.MarkDeprecated("insecure-bind-address", "This flag will be removed in a future version.")
 	fs.Lookup("insecure-bind-address").Hidden = false
 
-	fs.IntVar(&s.BindPort, "insecure-port", s.BindPort, ""+
+	fs.IntVar(&s.BindPort, "secure-port", s.BindPort, ""+
 		"The port on which to serve unsecured, unauthenticated access.")
-	fs.MarkDeprecated("insecure-port", "This flag will be removed in a future version.")
-	fs.Lookup("insecure-port").Hidden = false
+	fs.MarkDeprecated("secure-port", "This flag will be removed in a future version.")
```
Comment (MalloZup) [inline staging/src/k8s.io/apiserver/pkg/server/options/deprecated_insecure_serving.go]: i think here we will not remove this in future.

**Review by MalloZup — CHANGES_REQUESTED** [review]
hi @RA489 thank you for PR.

I added some nitpicks that we might need to consider, especially the port-numbers and some small concern about the doc and statements

:sunflower:  :santa: 
Ciao enjoy the day

File: cluster/centos/master/scripts/apiserver.sh (hunk 1 above)
Comment (yagonobre) [reply cluster/centos/master/scripts/apiserver.sh]: +1, We can use the default port `6443`
https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/

File: cmd/kubeadm/app/phases/controlplane/manifests.go [hunk 4]
```diff
@@ -120,7 +120,7 @@ func CreateStaticPodFiles(manifestDir string, cfg *kubeadmapi.InitConfiguration,
 func getAPIServerCommand(cfg *kubeadmapi.InitConfiguration) []string {
 	defaultArguments := map[string]string{
 		"advertise-address":               cfg.LocalAPIEndpoint.AdvertiseAddress,
-		"insecure-port":                   "0",
+		"secure-port":                   "0",
```
Comment (yagonobre) [inline cmd/kubeadm/app/phases/controlplane/manifests.go]: You can't set `secure-port` to 0. And `secure-port` was set in line 133, so you can just remove this line.

File: cmd/kubeadm/app/phases/controlplane/manifests_test.go [hunk 5]
```diff
@@ -152,7 +152,7 @@ func TestGetAPIServerCommand(t *testing.T) {
 			},
 			expected: []string{
 				"kube-apiserver",
-				"--insecure-port=0",
+				"--secure-port=0",
```
Comment (yagonobre) [inline cmd/kubeadm/app/phases/controlplane/manifests_test.go]: Use `fmt.Sprintf("%d", cfg.LocalAPIEndpoint.BindPort)`, in all cases.

File: cmd/kubeadm/app/phases/selfhosting/selfhosting_test.go [hunk 6]
```diff
@@ -54,7 +54,7 @@ spec:
     - --kubelet-client-certificate=/etc/kubernetes/pki/apiserver-kubelet-client.crt
     - --advertise-address=192.168.1.115
     - --requestheader-client-ca-file=/etc/kubernetes/pki/front-proxy-ca.crt
-    - --insecure-port=0
+    - --secure-port=0
```
Comment (yagonobre) [inline cmd/kubeadm/app/phases/selfhosting/selfhosting_test.go]: Use `fmt.Sprintf("%d", cfg.LocalAPIEndpoint.BindPort)`, in all cases.

**Review by yagonobre — COMMENTED** [review]
Thanks @RA489 for the PR and @MalloZup for the review!
I have added some comments.

**Review by neolit123 — COMMENTED** [review]
@RA489 
please leave only the cmd/kubeadm changes as part of this PR.
the rest of the changes are out of scope for us.

/priority backlog
/kind cleanup

File: cmd/kubeadm/app/phases/controlplane/manifests_test.go [hunk 7]
```diff
@@ -154,7 +154,7 @@ func TestGetAPIServerCommand(t *testing.T) {
 			},
 			expected: []string{
 				"kube-apiserver",
-				"--insecure-port=0",
+				"--secure-port=fmt.Sprintf("%d", cfg.LocalAPIEndpoint.BindPort)",
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/controlplane/manifests_test.go]: `Sprintf` is a number of times slower than `Itoa`, please use that and also ideally store it in a variable per function.

i think it's OK to change the instances in `manifests.go` too.

**Review by neolit123 — COMMENTED** [review]
thanks for the update @RA489 
added one comment about `Itoa`.

/ok-to-test

**Review by yagonobre — COMMENTED** [review]
/lgtm
ping @neolit123 to approver
/remove-sig apiserver

**Review by neolit123 — APPROVED** [review]
thanks
/approve

**Comment by liggitt** [comment]
This should be reverted... removing this means the insecure port is open by default

**Comment by dims** [comment]
cc @neolit123

**Comment by liggitt** [comment]
Revert open in https://github.com/kubernetes/kubernetes/pull/75425

**Pull request merged**